odoo/tools/misc.py

```diff
@@ -1132,6 +1132,15 @@ def html_escape(text):
     def html_escape(text):
         return werkzeug.utils.escape(text)
 
+def babel_locale_parse(lang_code):
+    try:
+        return babel.Locale.parse(lang_code)
+    except:
+        try:
+            return babel.Locale.default()
+        except:
```

Comment:
how can `babel.Locale.default()` produce an exception? In the weird scenario where a sysadmin has set an invalid code as default locale, it would just return `None` (`babel.Locale.parse(None)` is `None`)

Reply:
@mart-e 
```python
import babel
import os
for name in ['LANGUAGE', 'LC_ALL', 'LC_CTYPE']:
    os.environ[name] = ''
os.environ['LANG'] = 'kur'
babel.Locale.default()
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/odoo/.local/lib/python3.6/site-packages/babel/core.py", line 193, in default
    return cls.parse(locale_string)
  File "/home/odoo/.local/lib/python3.6/site-packages/babel/core.py", line 331, in parse
    raise UnknownLocaleError(input_id)
babel.core.UnknownLocaleError: unknown locale 'kur'

Reply:
yes on my system for example, I have `ku_TR` that is valid and `babel.core.default_locale()` just parse it and check that it is the good format without matching it to whats available:

https://github.com/python-babel/babel/blob/5afe2b2f11dcdd6090c00231d342c2e9cd1bdaab/babel/core.py#L1112-L1113

```py
 ipdb> get_locale_identifier(parse_locale('zefrançais_Imaginistant.BEST_ENCODING@petrodollar'))
'zefrançais_Imaginistant'
ipdb> parse(get_locale_identifier(parse_locale('zefrançais_Imaginistant.BEST_ENCODING@petrodollar')))
*** NameError: name 'parse' is not defined
```